tests/functional/syntax/exceptions/test_syntax_exception.py

```diff
@@ -107,5 +108,29 @@ def foo():
 
 
 @pytest.mark.parametrize("bad_code", fail_list)
-def test_syntax_exception(assert_compile_failed, get_contract, bad_code):
-    assert_compile_failed(lambda: get_contract(bad_code), SyntaxException)
+def test_syntax_exception(bad_code):
+    with pytest.raises(SyntaxException):
+        compile_code(bad_code)
+
+
+def test_bad_staticcall_keyword():
```

Comment:
The test contract itself is incomplete. Can we make sure we unit test only the introduced change here?

A correct (i.e. compileable) Vyper contract would look like this

```vy
from ethereum.ercs import IERC20
from ethereum.ercs import IERC20Detailed


@external
def foo():
    extcall IERC20(msg.sender).transfer(
        msg.sender,
        convert(staticcall IERC20Detailed(msg.sender).decimals(), uint256),
    )
```

so we could have the following here:

```vy
from ethereum.ercs import IERC20
from ethereum.ercs import IERC20Detailed


@external
def foo():
    extcall IERC20(msg.sender).transfer(
        msg.sender,
        convert(staticall IERC20Detailed(msg.sender).decimals(), uint256),
    )
```

Another example with nested `staticcall` could be:

```vy
from ethereum.ercs import IERC20
from ethereum.ercs import IERC20Detailed


@view
def bar() -> uint256:
    return staticcall IERC20(msg.sender).balanceOf(
        convert(staticall IERC20Detailed(msg.sender).decimals(), address)
    )
```

In the last example, if you replace `view` with `pure`, it will raise the `staticcall` first instead of:

```console
vyper.exceptions.StateAccessViolation: Cannot call a view function from a pure function
```

which is the expected behaviour IMO.